src/Type/ComponentBasedType.ts

```diff
@@ -0,0 +1,120 @@
+import { MOORING_PAYLOAD, VENDOR_PAYLOAD, WorkOrder_PAYLOAD } from "./ApiTypes"
+
+export interface CustomerDataProps {
+  customer: any
+  editMode: boolean
+  closeModal: () => void
+  getCustomer: () => void
+}
+
+export interface BoatData {
+  id: string
+  customerName: string
+  mooringServiceDate: string
+  mooringLocation: string
+}
+
+export interface BillsData {
+  workOrderNo: string
+  customerName: string
+  assignedTo: string
+  date: string
+}
+
+export interface LoginFormProps {
+  Label: string
+  typeEmail: string
+  typePass: string
+  showSinUp: boolean
+  admin?: boolean
+}
+
+export interface MooringDetail {
+  id: string
+  mainContact: string
+  mooringNumber: string
+  boatName: string
+}
+
+export interface RowExpansionProps {
+  mooringDetails: MooringDetail[]
+}
+
+export interface AddMooringProps {
+  moorings: MOORING_PAYLOAD
+  editMode: boolean
+}
+
+export interface Technician_Data {
+  id: string
+  techniciansName: string
+  openWorkOrders: string
+  completedJobs: string
+}
+
+export interface AddVendorProps {
+  vendors: VENDOR_PAYLOAD
+  editMode: boolean
+  closeModal: () => void
+  getVendor: () => void
+}
+
+export interface AccountPayableProps {
+  invoice: string
+  mooringid: string
+  name: string
+  technicianName: string
+  services: string
+  time: string
```

Comment:
AccountPayabale and AccountRecievable Props does not have any different can be declared single

Reply:
For now, these don't have any difference, but once the backend API is implemented, the props will likely change. That's why i'm keeping them separate for future modifications.

Reply:
then extract the common and use extend method